### PR TITLE
feat(targets): Accept overrides from callers to askFor methods

### PR DESCRIPTION
**Pull request by jskrzypek**

### CODE REVIEW

**Comment by jskrzypek** [comment]
Need to test interactions with generated gulp tasks from https://github.com/thaiat/generator-sublime before merging.

**Comment by jskrzypek** [comment]
Looks like This does not collide with generated tasks. Waiting for :shipit: to merge.

File: lib/targets.js [hunk 1]
```diff
@@ -117,8 +117,12 @@ var checkSingleTarget = exports.checkSingleTarget = function(args, opts) {
     return retval;
 };
 
-var askForTargets = function(taskname, defaultTargets, defaultMode, checkFn) {
-    taskname = taskname || '';
+var askForTargets = function(opts) {
+    var taskname = opts.taskname || '';
+    var checkFn = opts.multiple ? checkTargets : checkSingleTarget;
```
Comment (thaiat) [inline lib/targets.js]: pass directly the function here, if we have another one we don't need to change to logic

File: lib/targets.js [hunk 2]
```diff
@@ -135,6 +139,17 @@ var askForTargets = function(taskname, defaultTargets, defaultMode, checkFn) {
         .describe('m', chalk.cyan('Run the gulp task in the specified mode (prod or dev)'))
         .check(checkFn)
         .argv;
+
+    if(!_(opts.overrides).isEmpty()) {
```
Comment (thaiat) [inline lib/targets.js]: why do you need a overrides, just pass the options flat

File: lib/targets.js [hunk 3]
```diff
@@ -144,11 +159,24 @@ var askForTargets = function(taskname, defaultTargets, defaultMode, checkFn) {
...
  * @method askForMultipleTargets
  * @param {String} taskname - The name of the task that will accept multiple targets.
+ * @param {Object} overrides - (optional) Overrides the targets & mode from the command-line
  *
  * @returns {Object} - The yargs object
  */
-exports.askForMultipleTargets = function(taskname) {
-    return askForTargets(taskname, getAllTargets(), 'dev', checkTargets);
+exports.askForMultipleTargets = function(taskname, overrides) {
+    if(overrides && overrides.target) {
```
Comment (thaiat) [inline lib/targets.js]: use x || y instead of _.isEmpty

File: lib/targets.js [hunk 4]
```diff
@@ -156,9 +184,35 @@ exports.askForMultipleTargets = function(taskname) {
...
+    }
+    var opts = {
+        taskname: taskname || '',
+        multiple: false,
+        overrides: overrides
+    };
+    return askForTargets(opts);
 };
+
+// var opts = _({
```
Comment (thaiat) [inline lib/targets.js]: remove comments

File: test/mocha/targets.test.js [hunk 5]
```diff
@@ -25,15 +25,95 @@ describe('targets', function() {
     });
 
     describe('askFor()', function() {
-        it('askForMultipleTargets() should return object args', function() {
+        it('askForMultipleTargets() should return object args  with proper defaults', function() {
```
Comment (thaiat) [inline test/mocha/targets.test.js]: you should have to change existing test, just add missings

**Comment by jskrzypek** [comment]
@thaiat Can you take a look again? some of the complication for dealing with the `opts.target` came from the fact that I was trying to use logic that would work for both an array and a string, so I just ended up using `[].concat()` to make sure I got an array.

**Comment by thaiat** [comment]
ok you can merge

**Pull request merged**